lib/dry/logic/predicates.rb

```diff
@@ -197,6 +197,13 @@ def case?(pattern, input)
           pattern === input
         end
 
+        def uuid_v4?(input)
+          return false unless str?(input)
```

Comment:
This is not needed, predicates should check just one thing and assume the type is correct. In actual usage like `dry-schema` you'd have `required(:uuid).filled(:string, :uuid_v4?)`.

Reply:
Okay, I think that's fine. I'll update the PR to reflect that.